### DISCUSSION TITLE
User management question

### DISCUSSION THREAD

**Discussion by sergey-dme** [Budibase/budibase#1970] [Help] [answered]
Hi all,

Could someone please clarify on how user management works on portal-level and app-level. 
I mean - there is Users in builder/portal/manage/users - the main users table, and each app has Users table too and there's app screen _/users_
The Users tables are linked. But what **_/users_** in app used for? I can create new one in the app but it doesn't show up in Users table.
Can't figure out the philosophy/idea..

If it's already explained in other ticket or docs, I would thank for the link to it.

**Comment by mike12345567** [comment, marked as answer]
Hi @sergey-dme thanks for the question - the way this works is two parts as you've realised, there is two concepts of users, the global users and then app level user metadata.

The one issue of what you've described there is that its possible to create a screen which could create users, this is confusing and not really the designed functionality of Budibase, the only way to create global users is for them to be invited by an administrator (later this will also include group administrators).

The general idea with Budibase is that users are managed globally across the whole Budibase installation, so that within a company there is no need to keep re-inviting co-workers to join new apps, once they've been invited once they have access to the platform and then they just have to be granted access to the app itself (something that will be made a lot easier with the addition of groups).

The idea of the users table within an app is purely to store app specific metadata against the user, for example if we have an app that allows employees to request time off, you would want to store these time off requests with a relation to the user, but you wouldn't want to store this data globally. Instead we store some app specific metadata about the user which can then be used within the app, making sure that information that is specific to that app stays fully within the context of it.

I'll raise this with the team, the possibility of making it more clear that within an app its not a users table, but instead metadata stored against users that are part of the system - I will also look at removing the ability to create users screens as these would have no function.

**Reply by sergey-dme** [reply]
@mike12345567 Now it's more clear. Thank you for the explanation. 

Mb it's outside the concept, but it would be cool if admin has access to first/last name users fields. It would exclude needless for users to "setup his profile" after logging in. 
As for me, for users in my department/organization I'd like to block the option to change their first/last names

**Reply by mike12345567** [reply]
That makes sense, in one of the next releases this will be an option (the admin updating first/last name) I have already made this change myself.

Will speak to the team about adding an option to disable the users ability to edit information about themselves as I can imagine in a company there are several scenarios in which this would make sense.

**Reply by sergey-dme** [reply]
@mike12345567 Thanks Mike. 
Yeah, if we are talking about an app that being used as an organization internal tool to automate some process - organization fills up/pre-defines and controls user profile in most of the use cases. 
There are dozen of different tools in my company, all of them internally built, most of them are complex. And I can't change my name in any of them and it's no needed! Only employer can control my profile. The only tool where I can change my name/nickname is a HR tool - hub like a forum with group news, discussions, etc.
